io.go

```diff
@@ -31,3 +31,16 @@ func ReadSignalFile(path string, sampleRate float64) (*Signal, error) {
 	}
 	return &signal, scanner.Err()
 }
+// Function allows for reading arrays
+func ReadArray(dataArray []float64, sampleRate float64) (*Signal, error){
```

Comment:
Do you need to copy the array or can it be like 

```
signal := Signal{
		SampleRate: sampleRate,
		Signal:   dataArray,
	}
```

Reply:
yah instantiating it would work also!

Reply:
I would have to call the type from a different part of the application to update Signal var. I guess totally do able.

Reply:
Cool! If you make the change then Ill mergar this PR :)

Reply:
I am pretty sure i pushed to the right folder! Been a long week, sorry for the delay.

Reply:
I think something like this should work, 

```
func CreateSignalFromArray(signal []float64, sampleRate float64) *Signal {
	return &Signal{
		SampleRate: sampleRate,
		Signal:     signal,
	}
}
```

 - Replaced the return with &Signal instead of (&Signal, error) since there isn't any error returned
 - You can use the array directly, no need to copy the values. 
 - Changed to name "signal" since I think the go-lang best practice is not to have type name on variables (like dataArray). 
 
 This is using the same array as the one that is inserted, if you want to use a deep copy you can use something like 
 ```
 cpy := make([]float64, len(data))
copy(cpy, data)
```